torch/export/_trace.py

```diff
@@ -433,12 +444,21 @@ def _export_non_strict(
     with torch.nn.utils.stateless._reparametrize_module(
         mod, fake_params_buffers
     ), grad_safe_guard, _ignore_backend_decomps():  # type: ignore[attr-defined]
-        gm, graph_signature = transform(aot_export_module)(
-            mod,
-            (*fake_args, *fake_kwargs.values()),
-            trace_joint=False,
-            pre_dispatch=pre_dispatch,
-        )
+        try:
+            gm, graph_signature = transform(aot_export_module)(
+                mod,
+                (*fake_args, *fake_kwargs.values()),
+                trace_joint=False,
+                pre_dispatch=pre_dispatch,
+            )
+        except NonFakeInputError as e:
+            # TODO(zhxchen17) better error message.
+            raise _tensor_constants_error(None) from e
+
+    if len(list(gm.named_buffers())) > 0:
+        # torch.tensor() call.
+        # TODO(zhxchen17) better error message.
+        raise _tensor_constants_error(None)
```

Comment:
I think we shouldn't leave good error messages as a todo; it's like impossible for a user to tell where a tensor constant came from without a reasonable stack trace. 

1. For inline constants: shouldn't we have a stack trace on the get_attr node?
2. For module attributes: can we find the owning module in the hierarchy?